### PR TITLE
examples: Update redis example with `useOptimistic`

### PR DESCRIPTION
**Pull request by leerob**

### CODE REVIEW

File: examples/with-redis/app/form.tsx [hunk 1]
```diff
@@ -73,8 +76,8 @@ type FeatureState = {
 };
 
 export default function FeatureForm({ features }: { features: Feature[] }) {
-  const formRef = useRef<HTMLFormElement>(null);
-  const [state, mutate] = useOptimistic(
+  let formRef = useRef<HTMLFormElement>(null);
+  let [state, mutate] = useOptimistic(
```
Comment (brandonnorsworthy) [inline examples/with-redis/app/form.tsx:80]: Honest question here, Why change these from const to let? Is that the standard thoughout the repo?

Comment (leerob) [reply examples/with-redis/app/form.tsx:80]: It doesn't really matter, use whatever you prefer 😄

File: examples/with-redis/app/form.tsx [hunk 2]
```diff
@@ -112,6 +115,7 @@ export default function FeatureForm({ features }: { features: Feature[] }) {
     score: "1",
   };
   let saveWithNewFeature = saveFeature.bind(null, featureStub);
+  let [_pending, startTransition] = useTransition();
```
Comment (leerob) [inline examples/with-redis/app/form.tsx:118]:
```suggestion
  let [_, startTransition] = useTransition();
```

File: examples/with-redis/app/form.tsx [hunk 3]
```diff
@@ -23,22 +23,25 @@ function Item({
   pending: boolean;
   mutate: any;
 }) {
-  const upvoteWithId = upvote.bind(null, feature);
+  let upvoteWithId = upvote.bind(null, feature);
+  let [_pending, startTransition] = useTransition();
```
Comment (leerob) [inline examples/with-redis/app/form.tsx:27]:
```suggestion
  let [_, startTransition] = useTransition();
```

File: examples/with-redis/app/form.tsx [hunk 4]
```diff
@@ -23,22 +23,25 @@ function Item({
   pending: boolean;
   mutate: any;
 }) {
-  const upvoteWithId = upvote.bind(null, feature);
+  let upvoteWithId = upvote.bind(null, feature);
+  let [_, startTransition] = useTransition();
```
Comment (leerob) [inline examples/with-redis/app/form.tsx:27]:
```suggestion
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  let [pending, startTransition] = useTransition();
```

File: examples/with-redis/app/form.tsx [hunk 5]
```diff
@@ -112,6 +115,7 @@ export default function FeatureForm({ features }: { features: Feature[] }) {
     score: "1",
   };
   let saveWithNewFeature = saveFeature.bind(null, featureStub);
+  let [_, startTransition] = useTransition();
```
Comment (leerob) [inline examples/with-redis/app/form.tsx:118]:
```suggestion
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  let [pending, startTransition] = useTransition();
```

File: examples/with-redis/app/form.tsx [hunk 6]
```diff
@@ -23,22 +23,26 @@ function Item({
   pending: boolean;
   mutate: any;
 }) {
-  const upvoteWithId = upvote.bind(null, feature);
+  let upvoteWithId = upvote.bind(null, feature);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  let [pending, startTransition] = useTransition();
```
Comment (leerob) [inline examples/with-redis/app/form.tsx:28]:
```suggestion
  let [isPending, startTransition] = useTransition();
```

File: examples/with-redis/app/form.tsx [hunk 7]
```diff
@@ -112,6 +116,8 @@ export default function FeatureForm({ features }: { features: Feature[] }) {
     score: "1",
   };
   let saveWithNewFeature = saveFeature.bind(null, featureStub);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  let [pending, startTransition] = useTransition();
```
Comment (leerob) [inline examples/with-redis/app/form.tsx:120]:
```suggestion
  let [isPending, startTransition] = useTransition();
```

**Pull request merged**